Realm/RLMObservation.hpp

```diff
@@ -114,6 +114,9 @@ class RLMObservationInfo {
 
     template<typename F>
     void forEach(F&& f) const {
+        // Ensure that `this` continues to exist until the end of the function
+        // by retaining its owner
+        __attribute__((objc_precise_lifetime)) id self = object;
```

Comment:
We use the attribute directly everywhere else, and in this case the macro doesn't seem any clearer to me.